rdtools/normalization.py

```diff
@@ -11,6 +11,62 @@ class ConvergenceError(Exception):
     '''Rescale optimization did not converge'''
     pass
 
+def normalize_with_expected_power(pv, expected_power, irradiance, pv_input='power'):
+    '''
+    Normalize pv output based on expected PV power.
+
+    Parameters
+    ----------
+    pv : pd.Series
+        Right-labeled time series PV energy or power. If energy, should *not*
+        be cumulative, but only for preceding time step.
+    expected_power : pd.Series
+        Right-labeled time series of expected PV power.
+    irradiance : pd.Series
+        Right labeled time series of plane-of-array irradiance associated with `expected_power`
+    pv_input : str
+        'power' or 'energy' to specify type of input used for pv parameter
+
+    Returns
+    -------
+    normalized_energy : pd.Series
+        Energy normalized based on `expected_power
+    insolation : pd.Series
+        Insolation associated with each normalized point
+
+    '''
+    
+    freq = check_series_frequency(pv, 'pv')
+
+    if pv_input == 'power':
+        energy = energy_from_power(pv, freq)
+    elif pv_input == 'energy':
+        energy = pv.copy()
+        energy.name = 'energy_Wh'
+    else:
+        raise ValueError("Unexpected value for pv_input. pv_input should be 'power' of 'energy'.")
+
+    model_tds, mean_model_td = delta_index(expected_power)
+    measure_tds, mean_measure_td = delta_index(energy)
+
+    # Case in which the model is as or more frequent than the measurements
+    if mean_model_td <= mean_measure_td:
+
+        expected_energy = energy_from_power(expected_power, freq)
+        insolation = energy_from_power(irradiance, freq)
+
+    # Case in which the model less frequent than the measurements
+    elif mean_model_td > mean_measure_td:
+
+        expected_power = interpolate(expected_power, pv.index)
+        expected_energy = energy_from_power(expected_power, freq)
+
+        irradiance = interpolate(irradiance, pv.index)
+        insolation = energy_from_power(irradiance, freq)
```

Comment:
Both branches of this if/elif block call `insolation = energy_from_power(irradiance, freq)` and `expected_energy = energy_from_power(expected_power, freq)`, I think could change to:
```python
    if mean_model_td > mean_measure_td:
        expected_power = interpolate(expected_power, pv.index)
        irradiance = interpolate(irradiance, pv.index)

    expected_energy = energy_from_power(expected_power, freq)
    insolation = energy_from_power(irradiance, freq)
```
OK with me to leave as-is though

Reply:
Agreed, good catch